app/services/pipeline_report_service.rb

```diff
@@ -67,13 +67,39 @@ def call
   end
 
   def generate
-    pipeline_run = PipelineRun.find(@pipeline_run_id)
     if @pipeline_run_id.nil?
-      Rails.logger.error("Pipeline run doesn't exist.")
-      return ""
-    elsif pipeline_run.total_reads.nil? || pipeline_run.adjusted_remaining_reads.nil?
-      Rails.logger.error("Pipeline run #{@pipeline_run_id} has no reads.")
-      return ""
+      return JSON.dump(
+        pipelineRunInfo: {
+          pipelineRunComplete: false,
```

Comment:
What do you thing of restructuring this to a ```
status: oneof([COMPLETE, FAILED, WAITING])
errorType: oneof([USER, PIPELINE, UNKNOWN]),
statusMessage: "Error message"
```

Reply:
Combined `pipelineRunComplete` and `pipelineRunFailed` into `pipelineRunStatus`.

It looks like `known_user_error` is an existing field for pipeline runs, so I left that one as is for consistency. I didn't change `errorMessage` either because it's specific to errors, not all status messages.